core/promise/methods/noop/promise_issuer.go

```diff
@@ -23,21 +23,19 @@ import (
 	log "github.com/cihub/seelog"
 	"github.com/mysteriumnetwork/node/communication"
 	"github.com/mysteriumnetwork/node/core/promise"
+	"github.com/mysteriumnetwork/node/identity"
+	"github.com/mysteriumnetwork/node/money"
 	"github.com/mysteriumnetwork/node/service_discovery/dto"
 )
 
 const issuerLogPrefix = "[promise-issuer] "
 
-// NewPromiseIssuer creates instance of PromiseIssuer
-func NewPromiseIssuer(dialog communication.Dialog) *PromiseIssuer {
-	return &PromiseIssuer{
-		dialog: dialog,
-	}
-}
-
-// PromiseIssuer issues promises in such way, what no actual money is added to promise
+// PromiseIssuer issues promises in such way, that no actual money is added to promise
 type PromiseIssuer struct {
-	dialog communication.Dialog
+	IssuerID identity.Identity
+	Dialog   communication.Dialog
```

Comment:
Private attributes and factories. Caller will now know how to initiate all runtime variables

Reply:
Changed it to be a private.